app/views/main.html

```diff
@@ -68,7 +73,7 @@
         {{event.venue.name}}
       </td>
       <td>
-        <input type="checkbox" name="published" value="{{value}}" ng-click="togglePublished(event._id)" ng-checked="{{event.isPublished}}">
+        <input type="checkbox" name="published" value="{{value}}" ng-click="togglePublished(event._id);event.isPublished=!event.isPublished;" ng-checked="{{event.isPublished}}">
```

Comment:
Could also send the `$index` of the event to the `togglePublished` method and then update `$scope.events` based on success in the callback.